### PR TITLE
[ROX-14919] Scan images that are being tested

### PR DESCRIPTION
**Pull request by md2119**
## Description
Explicitly scan images to make sure the scan data is populated.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed
n/a

### CODE REVIEW

File: qa-tests-backend/src/test/groovy/DeploymentTest.groovy [hunk 1]
```diff
@@ -38,10 +38,14 @@ class DeploymentTest extends BaseSpecification {
 
     def setupSpec() {
         orchestrator.createDeployment(DEPLOYMENT)
+        ImageService.scanImage(DEPLOYMENT_IMAGE_NAME)
```
Comment (connorgorman) [inline qa-tests-backend/src/test/groovy/DeploymentTest.groovy:41]: Why is this necessary? the deployment creation should ensure that images are scanned. If this is not happening, then we should fix the root problem?

Comment (md2119) [reply qa-tests-backend/src/test/groovy/DeploymentTest.groovy:41]: I wonder if the resync changes have to do with the failures we have seen, including policy events not showing up. The coverage of this test is queried correctly. Although you raise a genuine concern, it doesn't seem to be the scope of this test.

If this fixes the failure, it is clear that the issue is not query framework or links etc.

**Review by charmik-redhat — APPROVED** [review]
